src/components/Requirements/SubRequirement.vue

```diff
@@ -138,46 +137,47 @@ type CompletedSubReqCourseSlot = {
 
 type IncompleteSubReqCourseSlot = {
   readonly isCompleted: false;
-  readonly courses: readonly CrseInfo[];
+  readonly courses: readonly FirestoreSemesterCourse[];
 };
 
 export type SubReqCourseSlot = CompletedSubReqCourseSlot | IncompleteSubReqCourseSlot;
 
 type Data = {
   showFulfillmentOptionsDropdown: boolean;
   displayDescription: boolean;
-  subReqFetchedCourseObjectsNotTakenArray: FirestoreSemesterCourse[];
-  dataReady: boolean;
   fulfilledSelfCheckCourses: FirestoreSemesterCourse[];
 };
 
+const generateSubReqIncompleteCourses = (
+  allTakenCourseIds: ReadonlySet<number>,
+  eligibleCourseIds: readonly number[]
+): readonly FirestoreSemesterCourse[] => {
+  const rosterCourses = eligibleCourseIds
+    .filter(courseID => !allTakenCourseIds.has(courseID))
+    .flatMap(courseID => fullCoursesJson[courseID] || []);
+  const subjectColors = allocateSubjectColors(new Set(rosterCourses.map(it => it.subject)));
+  return rosterCourses.map(rosterCourse =>
+    cornellCourseRosterCourseToFirebaseSemesterCourseWithCustomIDAndColor(
+      rosterCourse,
+      -1,
```

Comment:
Is the -1 here a dummy unique id for a course in the incomplete sub req course slot?

Reply:
yes, you can see in the clone function I added, I replace the id with a real allocated id when the course is dropped into a semester.